src/clue/adafruit_slideshow.py

```diff
@@ -0,0 +1,312 @@
+        return self._curr_img
+
+    @property
+    def order(self):
+        """Specifies the order in which the images are displayed. Options are random (``RANDOM``) or
+        alphabetical (``ALPHABETICAL``). Default is ``RANDOM``."""
+        return self._order
+
+    @order.setter
+    def order(self, order):
+        if order not in [PlayBackOrder.ALPHABETICAL, PlayBackOrder.RANDOM]:
+            raise ValueError("Order must be either 'RANDOM' or 'ALPHABETICAL'")
+
+        self._order = order
+        self._load_images()
+
+    @property
+    def brightness(self):
+        """Brightness of the backlight when an image is displaying. Clamps to 0 to 1.0"""
+        return self._brightness
+
+    @brightness.setter
+    def brightness(self, brightness):
+        if brightness < 0:
+            brightness = 0
+        elif brightness > 1.0:
+            brightness = 1.0
+        self._brightness = brightness
+
+    def update(self):
+        """Updates the slideshow to the next image."""
+        now = time.monotonic()
+        if not self.auto_advance or now - self._img_start < self.dwell:
+            return True
+
+        return self.advance()
+
+    def _get_next_img(self):
+
+        # handle empty queue
+        if not len(self.pic_queue):
+            if self.loop:
+                self._load_images()
+            else:
+                return ""
+
+        if self.direction == PlayBackDirection.FORWARD:
+            return self.pic_queue.popleft()
+        else:
+            return self.pic_queue.pop()
+
+    def _load_images(self):
+        dir_imgs = []
+        for d in self.dirs:
+            try:
+                new_path = os.path.join(self.folder, d)
+
+                # only add bmp imgs
+                if os.path.splitext(new_path)[1] == CONSTANTS.BMP_IMG_ENDING:
+                    dir_imgs.append(new_path)
+            except Image.UnidentifiedImageError as e:
+                continue
+
+        if not len(dir_imgs):
+            raise RuntimeError(CONSTANTS.NO_VALID_IMGS_ERR)
+
+        if self._order == PlayBackOrder.RANDOM:
+            shuffle(dir_imgs)
+        else:
+            dir_imgs.sort()
+
+        # convert list to queue
+        # (must be list beforehand for potential randomization)
+        self.pic_queue = collections.deque(dir_imgs)
+
+    def _advance_with_fade(self):
+
+        old_img = self._curr_img_handle
+        advance_sucessful = False
+
+        while not advance_sucessful:
+            new_path = self._get_next_img()
+            if new_path == "":
+                return False
+
+            try:
+                new_img = Image.open(new_path)
+
+                new_img = new_img.convert("RGBA")
+                new_img.putalpha(255)
+
+                new_img = new_img.crop(
+                    (0, 0, CONSTANTS.SCREEN_HEIGHT_WIDTH, CONSTANTS.SCREEN_HEIGHT_WIDTH)
+                )
+
+                if new_img.size[0] < 240 or new_img.size[1] < 240:

```

Comment:
should we use the constants instead of 240?

Reply:
yes!